### ISSUE TITLE
lts-5.9 (ghc 7.10.3) installation broken

### ISSUE THREAD

**Issue by kostmo** [commercialhaskell/stack#3801]
To reproduce the error:

Create a file "`testcase.hs`":

    #!/usr/bin/env stack
    -- stack --resolver lts-5.9 script

    main = putStrLn "Hello, World!"

Then:

    chmod +x testcase.hs
    ./testcase.hs

Failure observed:

    No information found for ghc-7.10.3.
    Supported versions for OS key 'linux64-nopie': GhcVersion 8.0.2, GhcVersion 8.2.1, GhcVersion 8.2.2

This failure was reproduced on a fresh Ubuntu 7.10 machine (64-bit) with stack version 1.6.3.

**Comment by kostmo** [comment]
Possibly related to this change:
https://github.com/fpco/stackage-content/pull/34/files#diff-287430ec2804b844873653582ccdc025L233

**Comment by kostmo** [comment]
Using (at this time, unreleased) Stack version 1.7.0 compiled from the master branch (git revision `2631803b9`) seems to fix this problem.

It does seem like a design issue, however, that builds using a specific, older version of stack (`1.3.2` in this case) would suddenly stop working.  Ideally builds would be eternally reproducible without forcing an upgrade of stack itself.

**Comment by kadoban** [comment]
@kostmo Yeah, in an ideal world that'd definitely be nice. I don't think it's quite possible right now though personally. Too many moving targets keep cropping up that stack needs to deal with just to keep old stuff working correctly.

**Comment by kostmo** [comment]
I've hit another problem now, this time when adding the `--compile` option, as in:

    stack -v --resolver lts-5.9 script --compile testcase.hs

This is now from using stack 1.7.0.
A large number of these errors appear in the output:

> ```relocation R_X86_64_32S against symbol `stg_bh_upd_frame_info' can not be used when making a shared object; recompile with -fPIC```

Here's a larger excerpt from the console output:
```
2018-01-23 17:09:40.234388: [debug] Found shared library libtinfo.so.5 in 'ldconfig -p' output
@(Stack/Setup.hs:585:29)
2018-01-23 17:09:40.234750: [debug] Did not find shared library libtinfo.so.6
@(Stack/Setup.hs:599:38)
2018-01-23 17:09:40.234834: [debug] Did not find shared library libncursesw.so.6
@(Stack/Setup.hs:599:38)
2018-01-23 17:09:40.234893: [debug] Found shared library libgmp.so.10 in 'ldconfig -p' output
@(Stack/Setup.hs:585:29)
2018-01-23 17:09:40.234966: [debug] Did not find shared library libgmp.so.3
@(Stack/Setup.hs:599:38)
2018-01-23 17:09:40.235012: [debug] Potential GHC builds: standard
@(Stack/Setup.hs:627:9)
2018-01-23 17:09:40.235067: [debug] Found already installed GHC builds: 
@(Stack/Setup.hs:456:13)
2018-01-23 17:09:40.549983: [debug] Trying to setup GHC build: standard
@(Stack/Setup.hs:950:9)
2018-01-23 17:09:40.551821: [info] Preparing to install GHC to an isolated location.
@(Stack/Setup.hs:884:5)
2018-01-23 17:09:40.552178: [info] This will not interfere with any system-level installation.
@(Stack/Setup.hs:893:5)
2018-01-23 17:09:40.553957: [debug] Downloading from https://github.com/commercialhaskell/ghc/releases/download/ghc-7.10.3-release/ghc-7.10.3b-x86_64-deb7-linux.tar.xz to /home/kostmo/.stack/programs/x86_64-linux/ghc-7.10.3.tar.xz ...

[...]

2018-01-23 17:18:11.561268: [debug] Build platform inferred as: x86_64-unknown-linux
@(Stack/Setup.hs:1090:54)
2018-01-23 17:18:11.566653: [debug] Host platform inferred as: x86_64-unknown-linux
@(Stack/Setup.hs:1090:54)
2018-01-23 17:18:11.572471: [debug] Target platform inferred as: x86_64-unknown-linux
@(Stack/Setup.hs:1090:54)
2018-01-23 17:18:11.599382: [debug] GHC build  : x86_64-unknown-linux
@(Stack/Setup.hs:1090:54)
2018-01-23 17:18:11.599613: [debug] GHC host   : x86_64-unknown-linux
@(Stack/Setup.hs:1090:54)
2018-01-23 17:18:11.599753: [debug] GHC target : x86_64-unknown-linux
@(Stack/Setup.hs:1090:54)

[...]

2018-01-23 17:25:55.224698: [debug] Run process within /home/kostmo/blah/: /home/kostmo/.stack/programs/x86_64-linux/ghc-7.10.3/bin/ghc -hide-all-packages -packagebase /home/kostmo/blah/testcase.hs
@(subs/rio/src/RIO/Process.hs:191:3)
/usr/bin/ld: /home/kostmo/blah/testcase.o: relocation R_X86_64_32S against symbol `stg_bh_upd_frame_info' can not be used when making a shared object; recompile with -fPIC
```

**Comment by mgsloan** [comment]
Certainly, ideally old versions wouldn't have bugs.  Since it's fixed in 1.7, closing this issue.

**Comment by borsboom** [comment]
I finally got my new build server ready and could tackle making some GHC bindists for 7.10.3 and 8.0.1 that are patched with `configure` scripts that detect GCC versions that need `-no-pie` or `--no-pie`.  See https://github.com/fpco/stackage-content/pull/36.
